pkg/discovery/dns/resolver.go

```diff
@@ -71,7 +72,10 @@ func (s *dnsSD) Resolve(ctx context.Context, name string, qtype QType) ([]string
 		}
 		ips, err := s.resolver.LookupIPAddr(ctx, host)
 		if err != nil {
-			return nil, errors.Wrapf(err, "lookup IP addresses %q", host)
+			dnsErr, ok := err.(*net.DNSError)
+			if !(s.resolverType == GolangResolverType && ok && dnsErr.IsNotFound) {
```

Comment:
Do we need to differentiate by resolver? I think it's enough that we cast to error type, no? (:

Reply:
Yes we do as explained here: https://github.com/thanos-io/thanos/issues/3186#issuecomment-700730106

The behaviour of the diff resolvers are different when the host does not exists.

Reply:
>I wonder if this makes sense, maybe putting it behind some option would do 🤔 

I disagree purely based on the diff behaviour of the resolvers when NXDOMAIN (or similar errors) happens.

Reply:
>By masking this error we will never know about this error right?
In this case we could also add warning logs so that the error is not totally masked, it just prevents a crash.

Reply:
`(*net.DNSError)` is never used in miekg so we can just check the cast error and add comment (which is exactly what you said): 

```
miekgdns and the default DNS server have different behaviors when a host can not be resolved. The default DNS resolved returns a err no such host while the miekgdns returns a nil response and a nil error.
```

(: 

Resolver type is not needed. WDYT?